### ISSUE TITLE
Custom runner is executed with bundled nodejs version instead of version installed locally (in cli)

### ISSUE THREAD

**Issue by jsifalda** [cypress-io/cypress#25293]
### Current behavior

Custom mocha runner is executed with nodejs version v16.16 (the same version is bundled in cypress)

### Desired behavior

The custom mocha runner should be executed with nodejs version 18.12.1 which is installed locally.

At least based on the change made in cypress v9 where the cypress should always use system nodejs version https://docs.cypress.io/guides/references/changelog#9-0-0. 

If not so, maybe there is not a bug but just a question, how to make sure the cypress will always use my nodejs version instead of bundled version?

### Test code to reproduce

Here is code to reproduce the issue, it can be reproduced in newly cypress repo (no special setup needed)
https://github.com/jsifalda/cypress-test-tiny

here is the proof: https://capture.dropbox.com/NUrUOWAncAFTAwtO
here is the simple mocha reporter code: https://capture.dropbox.com/wx1vmCLadU8dSFiZ

### Cypress Version

12.2.0

### Node version

this is the issue - cypress uses 16.16 instead of 18.12.1

### Operating System

Mac Ventura 13.1 (22C65)

### Debug Logs

```shell
~/Dropbox/dev/cypress-test-tiny (master*) » node -v && npx cypress -v &&  npx cypress run                                                        jsifalda@Jiris-MacBook-Pro-2020
v18.12.1
Cypress package version: 12.2.0
Cypress binary version: 12.2.0
Electron version: 21.0.0
Bundled Node version: 16.16.0

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        12.2.0                                                                         │
  │ Browser:        Electron 106 (headless)                                                        │
  │ Node Version:   v18.12.1 (/Users/jsifalda/.nvm/versions/node/v18.12.1/bin/node)                │
  │ Specs:          1 found (spec.cy.js)                                                           │
  │ Searched:       cypress/e2e/**/*.cy.{js,jsx,ts,tsx}                                            │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  spec.cy.js                                                                      (1 of 1)
---node version 16.16.0
```

**Comment by lmiller1990** [comment]
Confusingly, it says 

```
Node Version:   v18.12.1 (/Users/jsifalda/.nvm/versions/node/v18.12.1/bin/node)            
```

I wonder if there's different places we check the Node.js version.

**Comment by lmiller1990** [comment]
I logged here: https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/project-base.ts#L304

Goes here: https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/reporter.js#L532

Shows Node.js 16 - this must be the electron version. 

Looking at the docs https://docs.cypress.io/guides/references/configuration#Node-version

> The Node version is used in Cypress to:
> - Build files in the [supportFile](https://docs.cypress.io/guides/references/configuration#Folders-Files).
> - Execute code in the config file.

It is defaulting to `system` in `cypress.config.js` - I see 19.x on my machine. I *think* what you are seeing is the intended behavior - we use your system version for things like processing specs and running plugins, but the actual run process (so, the process where we spawn `mocha`, etc) is executed in Electron using the bundled version.

I guess that would mean customer reporters are limited to the features supported in Node.js 16. Is this causing you a specific problem or just something you noticed while working on a custom reporter?

**Comment by jsifalda** [comment]
Yep, I had an issue caused with nodejs version mishmash. Is there any way around it? I mean, to unify it? Even though it is meant to be that way, which I still doubt, it is pretty confusing. 
Plus, cypress documentation does not mention this caveat.

**Comment by lmiller1990** [comment]
We could update the docs to make this more explicit on this page: https://docs.cypress.io/guides/references/configuration#Node-version

This is currently how it works, we use our bundled Node.js version for the process. I don't know of any way to use the system node right now. What exactly are you doing in Node 18 that doesn't work in the bundled Node (v16)? If you can share your use case we can either

1) Recommend an alternative (or perhaps more idiomatic way) to do it
2) If it's a very compelling use case, bump this enhancement up in priority

Thanks!

**Comment by jsifalda** [comment]
Yes, please, update the docs so it is clear for everyone 👍 However, it is not the solution...

Well, since node v16, node fibers and all plugins which import c++ binaries stopped working like https://npmjs.com/deasync. 
So, after the update it breaks many things, including custom mocha reporters, which are used by cypress (in my case, it broke a reporter which needed to convert async code to sync code, so the mocha reporter finished the request before exiting as there is no other way around it, at least not known to me). 

I would really recommend unifying the nodejs versions, or at least providing a way around it... any better idea?

**Comment by lmiller1990** [comment]
I am not sure we can easily change the Node version electron uses to execute this code. We need to use the version of Node that is shipped with Electron to guarantee the product works as expected for all users.

Just to clarify, you expect your project to be on < Node.js 16 since you are utilizing fibers?

Which custom reporter are you using? it is open source? I wonder if we can find a way to make it work with Node.js 16+?

**Comment by jsifalda** [comment]
well, does it make sense to provide "system" provider for node version detection if you are not able to ensure it?

**Comment by lmiller1990** [comment]
Sorry, I'm not understanding your question -- what's the proposal?

**Comment by jsifalda** [comment]
I am only saying, it is not good to have this option: `nodeVersion: 'system'`, if we are not able to ensure cypress and all its parts be running in system's node version.

**Comment by lmiller1990** [comment]
I think it still has some usage, for example you probably want your preprocessor pipeline, building your supportFile, and executing your plugins to run using your local node version, so it's definitely a useful feature, even in the current state. 

Those things run in a different process to the actual Cypress electron app, which handles the test runtime, and things like printing the output of the tests with a custom reporter, which is where you ran into a problem.

I think we should make the docs more clear. It also might be possible to default to system for the Cypress process, too, but I am not sure if there's any implications, and it would likely be breaking change, so we'd need to consider that, too. I am not sure of the scope of this change.

In the meantime, to unblock you, did you find an alternative work-around? Are you able to share the custom reporter that isn't working for you, or is that closed source?
